cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -1507,12 +1507,12 @@ endif()
 if(tmva AND cuda AND tmva-gpu)
   message(STATUS "Looking for CUDA for optional parts of TMVA")
 
-  if(cxx11)
+  if(CMAKE_CXX_STANDARD EQUAL 11)
     find_package(CUDA 7.5)
-  elseif(cxx14)
+  elseif(CMAKE_CXX_STANDARD EQUAL 14)
     message(STATUS "Detected request for c++14, requiring minimum version CUDA 9.0 (default 7.5)")
     find_package(CUDA 9.0)
-  elseif(cxx17)
+  elseif(CMAKE_CXX_STANDARD EQUAL 17)
```

Comment:
Can we now make this `else()`?

Reply:
Yes, maybe. This changes depending on the version of CUDA and the host compiler. Maybe it's best to use a `try_compile()` in the future to check if things actually work and error out if not.

Reply:
yes, we should really on try_compile for most things .. as long as they get rerun/recalculated for every relevant changes (for example when the user change the requested C++ standard)